src/Symfony/Component/Validator/Mapping/Factory/LazyLoadingMetadataFactory.php

```diff
@@ -162,4 +166,45 @@ public function hasMetadataFor($value)
 
         return class_exists($class) || interface_exists($class, false);
     }
+
+    private function readFromCache(string $class): ?ClassMetadata
+    {
+        if (null === $this->cache) {
+            return null;
+        }
+        if ($this->cache instanceof CacheInterface) {
+            return $this->cache->read($class) ?: null;
+        }
+
+        return $this->cache->getItem($this->escapeClassName($class))->get();
+    }
+
+    private function writeToCache(ClassMetadata $metadata): void
+    {
+        if (null === $this->cache) {
+            return;
+        }
+        if ($this->cache instanceof CacheInterface) {
+            $this->cache->write($metadata);
+
+            return;
+        }
+
+        $item = $this->cache->getItem($this->escapeClassName($metadata->getClassName()));
```

Comment:
I missed this part: here, we fetch a second time from the backend. This should be avoided.

Reply:
That was the status quo with `Psr6Cache`.

Okay, let's try this again. I'm not throwing away the `CacheItem` this time. 🤓 What started as a refactoring task might now be a performance improvement. 😃